borsh-derive/src/internals/serialize/enums/mod.rs

```diff
@@ -177,18 +219,32 @@ fn process_field(
 ) -> syn::Result<()> {
     let parsed = field::Attributes::parse(&field.attrs)?;
 
-    let needs_bounds_derive = parsed.needs_bounds_derive(BoundType::Serialize);
+    let needs_bounds_derive = parsed.needs_bounds_derive::<IS_ASYNC>(BoundType::Serialize);
     generics
         .overrides
-        .extend(parsed.collect_bounds(BoundType::Serialize));
+        .extend(parsed.collect_bounds::<IS_ASYNC>(BoundType::Serialize));
 
     let field_variant_header = field_id.enum_variant_header(parsed.skip);
     if let Some(field_variant_header) = field_variant_header {
         output.header.extend(field_variant_header);
     }
 
     if !parsed.skip {
-        let delta = field_id.serialize_output(cratename, parsed.serialize_with);
+        let delta = field_id.serialize_output::<IS_ASYNC>(
+            &field.ty,
+            cratename,
+            if IS_ASYNC {
+                cfg_if! {
+                    if #[cfg(feature = "async")] {
+                        parsed.serialize_with_async
+                    } else {
+                        None
+                    }
+                }
+            } else {
+                parsed.serialize_with
+            },
```

Comment:
this looks like it can be extracted to a method `get_ser_with`, similar to `internals::attributes::field::Attributes::get_bounds`